### ISSUE TITLE
The JavaLibraryPlugin must not be applied for project/module that is creating an executable artifact

### ISSUE THREAD

**Issue by MartinAhrer** [cloudflightio/autoconfigure-gradle-plugin#127]
By default the JavaLibraryPlugin is applied to all modules (the project). The JavaLibraryPlugin only should be applied when producing a shareable library JAR.
For an executable artifact it is sufficient to apply the JavaPlugin or in addition the application/SpringBoot plugin.

Having the JavaLibraryPlugin applied by default causes the Spring Boot native image build to try producing a dynamic link library instead of producing an executable.

Below is a first attempt to fix that

```
class JavaConfigurePlugin : Plugin<Project> {
    override fun apply(project: Project) {
        if (project.name.endsWith("-server")) { // TODO figure out how to use isServer()
            project.plugins.apply(JavaPlugin::class)
        } else {
            project.plugins.apply(JavaLibraryPlugin::class)
        }
...

```

**Comment by klu2** [comment]
that is already merged since a couple of versions, forgot to close

**Comment by haraldradi** [comment]
now the "api" configuration is not available anymore: https://docs.gradle.org/current/userguide/java_library_plugin.html#sec:java_library_separation

**Comment by haraldradi** [comment]
Also see https://github.com/cloudflightio/autoconfigure-gradle-plugin/issues/195

**Comment by klu2** [comment]
Yes, as of #127 and  #137 we decided to not apply the JavaLibraryPlugin automatically anymore at least in applicationBuilds. As we can only evaluate that property afterEvaluate, we decided to not apply the JavaLibraryPlugin at all any more. In case you need it, please apply it manually.

The problematic code here is that one: https://github.com/cloudflightio/autoconfigure-gradle-plugin/blob/main/src/main/kotlin/io/cloudflight/gradle/autoconfigure/java/JavaConfigurePlugin.kt#L69

We need to wait for afterEvaluate, and that's too late to apply the plugin.

**Comment by haraldradi** [comment]
This is quite a breaking change.

**Comment by rspiegl** [comment]
The breaking change was marked with a new minor version because there wasn't a 1.0.0 released at that time. See https://github.com/cloudflightio/autoconfigure-gradle-plugin/releases/tag/0.9.0. In these release notes it is also written what to do if you still need the `java-library` plugin: `If you need the api dependency scope, then please apply the java-library plugin in your modules yourself`. 
For better clarity I've updated the documentation in https://github.com/cloudflightio/autoconfigure-gradle-plugin/pull/196. 
With this I kindly ask you to close this issue or state a reason to keep this issue opened.
